### ISSUE TITLE
User-submitted Bug Report [ryukuzui]

### ISSUE THREAD

**Issue by avluis** [csaki/Hentoid#37]
This is a user generated bug report provided upon request - [comment chain](http://fakkudroid.blogspot.com/2015/08/hentoid-v109.html?showComment=1440318850115#c187903389673922004).

User reports:
"Pururin downloads don't work; they just keep saying the 'Unhandled error reading image URLs' message. If it doesn't say that, the download stays at 0% forever."

A quick look at the generated bug report doesn't throw any flags and so I have opened this issue for more eyes to be able take a look at it - please note that bug reports can contain user data so access to this report will be done per request.

@csaki @neosubmarino you have [access](https://drive.google.com/folderview?id=0BycOIcYbBvF_RHlnU0hKM0JueDA&usp=sharing).

**Comment by neosubmarino** [comment]
Because there are no notable flags in the log I believe the issue isn't app-side but is actually server side. The fact that it only happens for pururin downloads reinforces this hypothesis. Cloudflare has also been known to have regional bias so the "Server did not respond" cloudflare errors may happen more often on other parts of the world.

This issue is related to Issue #15 and I believe fixing that along with automated reports when it happens, will take us 1 step closer to fixing this.

**Comment by avluis** [comment]
OK - after a few emails (and a bit of research) I was able to come up with a way to fix his specific issue.
Since the issue was specific to downloads from the Pururin service - and not from Fakku and Hitomi - I suspected that since a few commits were pushed to fix that particular issue (see commit 0282b1fb68660f2138f1c031692a747c2741d65e & issue #16 ).
Release 1.0.9 was issued integrating these commits but if you had a download already in progress and hanged (due to the download manager not being able to grab the special cookie with fix implemented on 1.0.9) then only removing these downloads and grabbing them again from Pururin would fix this particular issue.

In conclusion, this is a holdover bug for not canceling downloads from an older client.
Fix: Cancel the download and retry after installing the latest version of [Hentoid](https://github.com/csaki/Hentoid/releases/latest).

**Comment by davidkusuma911** [comment]
Hentoid crash after selecting item. usualy if i select more than 10 item. and how to delete custom group without deleting its content? hope you also add "select all" feature in the next update.

sorry if i post on the wrong label. im new in github🙏
